### PR TITLE
Implement CPM core domain with scheduling algorithm

### PR DESCRIPTION
**Pull request by nbrengle**
Add Critical Path Method (CPM) domain model following DDD principles:

Domain Model:
- Value objects: ActivityName, Duration, Float, UnscheduledActivity, ScheduledActivity, UnscheduledProjectNetwork, ScheduledProjectNetwork
- Domain service: schedule() implementing forward/backward pass CPM algorithm

Key Design Decisions:
- Use Decimal for precision instead of float
- Separate unscheduled/scheduled activities (no optionals)
- Immutable value objects with frozen dataclasses
- Network validation (cycles, dependencies, uniqueness, non-empty)
- ActivityName instead of ActivityId to signal value object semantics
- Activities use 'name' attribute instead of 'id' for clarity
- ActivityName supports string comparison for readable test assertions
- InitVar pattern allows passing string dependencies: depends_on=frozenset(["A", "B"])

Testing:
- 46 tests achieving 99% code coverage
- 7 property-based tests using hypothesis
- 5 correctness tests with hand-calculated CPM values
- Edge cases: diamond graphs, multiple critical paths, disconnected subgraphs

Infrastructure:
- GitHub Actions CI workflow
- Ruff linting with line-length=120, all rules enabled
- Mypy strict mode type checking
- Pytest with coverage threshold

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: parade/domain/project_network.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    """Value object representing a validated network of scheduled activities."""
+
+    @property
+    def project_duration(self) -> Duration:
+        """Calculate the total project duration.
+
+        Project duration is the maximum earliest finish time across all activities.
+        This represents the minimum time required to complete the entire project.
+        """
+        return max(activity.earliest_finish for activity in self.activities)
```
Comment (nbrengle) [inline parade/domain/project_network.py:125]: Non-blocking: I'm not sure that these are real domain concepts or clearer than just working with ProjectNetwork[ScheduledActivity] and ProjectNetwork[UnscheduledActivity] but I'm content with this approach for now.
